prov/sockets/src/sock_ep_msg.c

```diff
@@ -378,9 +381,9 @@ static int sock_ep_cm_getpeer(struct fid_ep *ep, void *addr, size_t *addrlen)
 	}
 
 	sock_ep = container_of(ep, struct sock_ep, ep);
-	*addrlen = MIN(*addrlen, sizeof(struct sockaddr_in));
-	memcpy(addr, sock_ep->dest_addr, *addrlen);
-	return (*addrlen == sizeof(struct sockaddr_in)) ? 0 : -FI_ETOOSMALL;
+	len = MIN(*addrlen, sizeof(struct sockaddr_in));
+	memcpy(addr, sock_ep->dest_addr, len);
+	return (len == sizeof(struct sockaddr_in)) ? 0 : -FI_ETOOSMALL;
```

Comment:
This function does not set addrlen.  Also, does memcpy handle len = 0?  If not, you'll need something like what you have for getname.  If so, then getname could be updated to match this flow.